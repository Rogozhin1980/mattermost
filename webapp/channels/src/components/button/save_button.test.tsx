// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import {shallow} from 'enzyme';
import React from 'react';

import SaveButton from './save_button';

describe('components/SaveButton', () => {
    const baseProps = {
        saving: false,
    };

    test('should match snapshot, on defaultMessage', () => {
        const wrapper = shallow(<SaveButton {...baseProps}/>);

        expect(wrapper).toMatchSnapshot();

        wrapper.setProps({defaultMessage: 'Go'});
        expect(wrapper).toMatchSnapshot();
    });

    test('should match snapshot, on savingMessage', () => {
        const props = {...baseProps, saving: true, disabled: true};
        const wrapper = shallow(<SaveButton {...props}/>);

        expect(wrapper).toMatchSnapshot();

        wrapper.setProps({savingMessage: 'Saving Config...'});
        expect(wrapper).toMatchSnapshot();
    });

    test('should match snapshot, extraClasses', () => {
        const props = {...baseProps, extraClasses: 'some-class'};
        const wrapper = shallow(<SaveButton {...props}/>);

        expect(wrapper).toMatchSnapshot();
    });
});
