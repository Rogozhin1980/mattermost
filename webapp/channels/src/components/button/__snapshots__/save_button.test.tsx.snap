// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/SaveButton should match snapshot, extraClasses 1`] = `
<Button
  buttonType="submit"
  disabled={false}
  label={
    Object {
      "defaultMessage": "Save",
      "id": "save_button.save",
    }
  }
  loading={false}
  testId="saveSetting"
/>
`;

exports[`components/SaveButton should match snapshot, on defaultMessage 1`] = `
<Button
  buttonType="submit"
  disabled={false}
  label={
    Object {
      "defaultMessage": "Save",
      "id": "save_button.save",
    }
  }
  loading={false}
  testId="saveSetting"
/>
`;

exports[`components/SaveButton should match snapshot, on defaultMessage 2`] = `
<Button
  buttonType="submit"
  disabled={false}
  label="Go"
  loading={false}
  testId="saveSetting"
/>
`;

exports[`components/SaveButton should match snapshot, on savingMessage 1`] = `
<Button
  buttonType="submit"
  disabled={true}
  label={
    Object {
      "defaultMessage": "Saving",
      "id": "save_button.saving",
    }
  }
  loading={true}
  testId="saveSetting"
/>
`;

exports[`components/SaveButton should match snapshot, on savingMessage 2`] = `
<Button
  buttonType="submit"
  disabled={true}
  label="Saving Config..."
  loading={true}
  testId="saveSetting"
/>
`;
